libfreerdp/core/nla.c

```diff
@@ -1234,9 +1319,9 @@ static BOOL nla_read_ts_credentials(rdpNla* nla, PSecBuffer ts_credentials)
 	ret = ber_read_sequence_tag(s, &length) &&
 	      /* [0] credType (INTEGER) */
 	      ber_read_contextual_tag(s, 0, &length, TRUE) &&
-	      ber_read_integer(s, NULL) &&
-	      /* [1] credentials (OCTET STRING) */
-	      ber_read_contextual_tag(s, 1, &length, TRUE) &&
+	      ber_read_integer(s, NULL);
+	/* [1] credentials (OCTET STRING) */
+	ret += ber_read_contextual_tag(s, 1, &length, TRUE) &&
 	      ber_read_octet_string_tag(s, &ts_password_creds_length) &&
```

Comment:
Should be &= here

Reply:
Hum in fact, the original sequence of instructions with & was better because we stop the treatment as soon as something is wrong, your changes break that.